### ISSUE TITLE
Add support for bib2gls .bib files (Latex glossaries-extra)

### ISSUE THREAD

**Issue by jschwe** [valentjn/vscode-ltex#211]
**Is your feature request related to a problem? Please describe.**
Currently there is no spell-checking for .bib files. The Latex package glossaries-extra provides `bib2gls`, a tool which uses 
.bib files to maintain glossary entries. It would be great to have spellchecking for the descriptions of the entries.

A sample bib2gls .bib looks like this:
```
@entry{some-label,
  name = {Some Name},
  description = {This can be a multi-line description.}
}

@entry{some-label2,
  name = {Some Name},
  description = {This can be a multi-line description.
  },
}

@abbreviation{some-label3,
  short = {shortform},
  long  = {longform},
 } 
```

**Describe the solution you'd like**
Entries can contain custom keys, but for me it would be enough to have spellchecking for the description key. 
Alternatively spellchecking for all keys would be even better, if there is an option to blacklist some keys (e.g. `see` and `see-also`, which expect a label).

**Describe alternatives you've considered**
The alternative I've been using is converting the compiled PDF to Word and running a spellchecker on the Glossary in Word.

Also thanks for your great extension, it's already been pretty helpful.

**Comment by valentjn** [comment]
Thanks for the great feature request. This is now implemented.

If there are any additional fields apart from `see` and `see-also` that should be ignored by default, let me know.

**Comment by valentjn** [comment]
Fix released in 8.3.0.

**Comment by jschwe** [comment]
Thank you for implementing this! It's working great and already caught some mistakes!
Some things I noticed:
- I misremembered the key name for see-also. The key is defined as `seealso` in glossaries-extra. Sorry about that.
- Is it possible to suppress `LTeX - UPPERCASE_SENTENCE_START` by default or as an option for the glossary? Glossaries provides the `\Gls*` variants to make the first letter uppercase if required, while `\gls*` does not change casing. Currently, I get a lot of false positives with this warning in the glossary for lowercase entry names.


Additional keys that (I use and) are defined in glossaries-extra by default and contain labels:
- `parent` (Used to define a parent entry for hierarchical glossaries)
- `category` (Used for category specific style modifications)

**Comment by valentjn** [comment]
`seealso` is fixed now. `UPPERCASE_SENTENCE_START` is now disabled in BibT<sub>E</sub>X files. `category` and `parent` are now ignored by default.

FWIW, there's a new setting [`ltex.bibtex.fields`](https://valentjn.github.io/vscode-ltex/docs/settings.html#ltexbibtexfields) that allows to ignore the values of additional fields (or override the defaults and enable a field again). If they are fields that might be ignored by many users, it's better to have them added to the default list, so thanks for the suggestions.

**Comment by valentjn** [comment]
Fix released in 8.4.0.
